feat: list-view click and drag on check box to select multiple rows

Ponnusamy1-V opened the pull request:
In list view click and drag on the check box to select multiple rows

click and drag will select
again click and drag on selected will unselect

![dragselect](https://user-images.githubusercontent.com/95607086/227536316-1e41e8f9-a35b-47a7-afe5-755d3df16c1c.gif)

>no-docs

Ponnusamy1-V committed:
feat: list-view click and drag on check box to select multiple rows

Ponnusamy1-V committed:
Merge branch 'develop' into develop

barredterra reviewed (commented):
Tested, works as expected. 👍🏼 

Could you add a cypress test to cover this new functionality, as well as documentation?

Not necessary, but maybe this could also be useful on report rows?

Ponnusamy1-V committed:
Update list_view.js

Remove unwanted code

Ponnusamy1-V committed:
Merge branch 'frappe:develop' into develop

Ponnusamy1-V committed:
Merge branch 'frappe:develop' into develop

nextchamp-saqib committed:
chore: add insights banner (#20487)

* chore: add insights banner

* chore: cleanup message and title

[skip ci]

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: date field shouldn't be formatted for TZ (#20486)

Date fields aren't timezone aware.

marination committed:
feat: Disable Sharing globally (#20318)

* feat: Disable Sharing globally

- Checkbox in System Settings
- If disabled, avoid share UI render
- Share APIs return None (non-obstructing) if share APIs are invoked

* feat: Settings checkbox must toggle share permission globally

- Treat feature like a perm toggler. Essentially noone is allowed to explicity share anything
- Implicit sharing via `ignore_share_permissions` is allowed. Devs can decide where sharing should happen under the hood
- UI is made read only and not hidden. Users must see who doc is already shared with
- Make sure perm APIs used by share feature return false if sharing is disabled
- Rename checkbox to `Disable Document Sharing`

* test: (server side) Impact of disabling sharing on APIs

- Also, fix missed system setting rename in `assign_to`

* fix: Inform assigner if assignee lacks perms and sharing is disabled

- misc: readable conditions

* fix: throw instead of msgprint

* fix: Typo and appropriate message for `throw`

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
test: fix test case to modified behaviour

We throw instead of showing warning now

barredterra committed:
test: switch tests to supported methods (#20494)

* fix: switch tests to supported methods

get_fetch_fields, update_linked_doctypes

* test: semantic assertions

* test: fixup deprecation tests imports

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: nestedset rename (#20498)

barredterra committed:
feat(contact template): clickable email ids and phone numbers, less labels (#20247)

* refactor(contact template): use for .. of loop

* refactor(contact template): clickable phone numbers and email ids

* refactor(contact template): less labels

* fix: escape phone and email ids

Ponnusamy1-V committed:
List view: toggle check boxes instead of left and right clicks

Ponnusamy1-V committed:
Cypress test case for list view drag select

Ponnusamy1-V committed:
Merge branch 'develop' into develop

Ponnusamy1-V committed:
added comments for drag select

Ponnusamy1-V commented:
> Tested, works as expected. 👍🏼
> 
> Could you add a cypress test to cover this new functionality, as well as documentation?
> 
> Not necessary, but maybe this could also be useful on report rows?

added comments and test cases

Ponnusamy1-V committed:
Update list_view_drag_select.js

Ponnusamy1-V committed:
Merge branch 'develop' into develop

Ponnusamy1-V commented:
> Tested, works as expected. 👍🏼
> 
> Could you add a cypress test to cover this new functionality, as well as documentation?
> 
> Not necessary, but maybe this could also be useful on report rows?

do you want me to update the documentation here, https://frappeframework.com/docs/v13/user/en/api/list ?

Ponnusamy1-V committed:
Merge branch 'develop' into develop

shariquerik reviewed (changes requested):
package.json changes are not needed you can remove them.

shariquerik committed:
Merge branch 'develop' into develop

shariquerik commented:
> do you want me to update the documentation here, [frappeframework.com/docs/v13/user/en/api/list](https://frappeframework.com/docs/v13/user/en/api/list) ?

You can add it over here https://frappeframework.com/docs/v14/user/en/hidden-features/desk

Ponnusamy1-V committed:
Update package.json

removed unwanted

Ponnusamy1-V commented:
> package.json changes are not needed you can remove them.

removed

Ponnusamy1-V commented:
> > do you want me to update the documentation here, [frappeframework.com/docs/v13/user/en/api/list](https://frappeframework.com/docs/v13/user/en/api/list) ?
> 
> You can add it over here https://frappeframework.com/docs/v14/user/en/hidden-features/desk

updated

Ponnusamy1-V committed:
Merge branch 'develop' into develop

shariquerik committed:
Merge branch 'develop' into develop

shariquerik committed:
chore: linter fix

shariquerik committed:
chore: linter fix

shariquerik committed:
Merge branch 'develop' into develop

shariquerik committed:
chore: linter fix

shariquerik committed:
chore: linter fix

shariquerik commented:
Unrelated test failing
